2024-07-11 12:53:28.432622: W tensorflow/stream_executor/platform/default/dso_loader.cc:55] Could not load dynamic library 'cudart64_100.dll'; dlerror: cudart64_100.dll not found
2024-07-11 12:53:28.432796: I tensorflow/stream_executor/cuda/cudart_stub.cc:29] Ignore above cudart dlerror if you do not have a GPU set up on your machine.
INFO flwr 2024-07-11 12:53:37,869 | app.py:165 | Starting Flower server, config: ServerConfig(num_rounds=10, round_timeout=None)
INFO flwr 2024-07-11 12:53:37,968 | app.py:179 | Flower ECE: gRPC server running (10 rounds), SSL is disabled
INFO flwr 2024-07-11 12:53:37,968 | server.py:89 | Initializing global parameters
INFO flwr 2024-07-11 12:53:37,968 | server.py:276 | Requesting initial parameters from one random client
INFO flwr 2024-07-11 12:53:39,311 | server.py:280 | Received initial parameters from one random client
INFO flwr 2024-07-11 12:53:39,312 | server.py:91 | Evaluating initial parameters
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:39,329 | server.py:98 | initial parameters (loss, other metrics): 0.6931471805599453, {'accuracy': 0.4033333333333333}
INFO flwr 2024-07-11 12:53:39,329 | server.py:104 | FL starting
DEBUG flwr 2024-07-11 12:53:39,758 | server.py:227 | fit_round 1: strategy sampled 2 clients (out of 3)
DEBUG flwr 2024-07-11 12:53:39,790 | server.py:241 | fit_round 1 received 2 results and 0 failures
WARNING flwr 2024-07-11 12:53:39,792 | fedavgm.py:198 | No fit_metrics_aggregation_fn provided
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:39,799 | server.py:131 | fit progress: (1, 0.4309062369547472, {'accuracy': 0.79}, 0.47024889999999964)
DEBUG flwr 2024-07-11 12:53:39,800 | server.py:178 | evaluate_round 1: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:53:39,832 | server.py:192 | evaluate_round 1 received 3 results and 0 failures
WARNING flwr 2024-07-11 12:53:39,833 | fedavg.py:273 | No evaluate_metrics_aggregation_fn provided
DEBUG flwr 2024-07-11 12:53:39,833 | server.py:227 | fit_round 2: strategy sampled 3 clients (out of 3)
DEBUG flwr 2024-07-11 12:53:39,870 | server.py:241 | fit_round 2 received 3 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:39,880 | server.py:131 | fit progress: (2, 0.43567787449586104, {'accuracy': 0.79}, 0.5507112000000003)
DEBUG flwr 2024-07-11 12:53:39,880 | server.py:178 | evaluate_round 2: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:39,920 | server.py:192 | evaluate_round 2 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:53:39,920 | server.py:227 | fit_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:39,959 | server.py:241 | fit_round 3 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:39,970 | server.py:131 | fit progress: (3, 0.44566503187687334, {'accuracy': 0.7866666666666666}, 0.6401073000000004)
DEBUG flwr 2024-07-11 12:53:39,970 | server.py:178 | evaluate_round 3: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:39,997 | server.py:192 | evaluate_round 3 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:53:39,997 | server.py:227 | fit_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,030 | server.py:241 | fit_round 4 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:40,041 | server.py:131 | fit progress: (4, 0.445645019378396, {'accuracy': 0.7866666666666666}, 0.7115802000000002)
DEBUG flwr 2024-07-11 12:53:40,041 | server.py:178 | evaluate_round 4: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,067 | server.py:192 | evaluate_round 4 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:53:40,068 | server.py:227 | fit_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,124 | server.py:241 | fit_round 5 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:40,135 | server.py:131 | fit progress: (5, 0.44564502862649996, {'accuracy': 0.7866666666666666}, 0.8058437000000005)
DEBUG flwr 2024-07-11 12:53:40,136 | server.py:178 | evaluate_round 5: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,160 | server.py:192 | evaluate_round 5 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:53:40,160 | server.py:227 | fit_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,205 | server.py:241 | fit_round 6 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:40,215 | server.py:131 | fit progress: (6, 0.44564502863521266, {'accuracy': 0.7866666666666666}, 0.8863187999999997)
DEBUG flwr 2024-07-11 12:53:40,216 | server.py:178 | evaluate_round 6: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,237 | server.py:192 | evaluate_round 6 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:53:40,237 | server.py:227 | fit_round 7: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,280 | server.py:241 | fit_round 7 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:40,292 | server.py:131 | fit progress: (7, 0.4456450286352385, {'accuracy': 0.7866666666666666}, 0.9629104000000002)
DEBUG flwr 2024-07-11 12:53:40,292 | server.py:178 | evaluate_round 7: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,319 | server.py:192 | evaluate_round 7 received 4 results and 0 failures
DEBUG flwr 2024-07-11 12:53:40,319 | server.py:227 | fit_round 8: strategy sampled 4 clients (out of 4)
DEBUG flwr 2024-07-11 12:53:40,359 | server.py:241 | fit_round 8 received 4 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:40,371 | server.py:131 | fit progress: (8, 0.4456450286352386, {'accuracy': 0.7866666666666666}, 1.0418015999999994)
DEBUG flwr 2024-07-11 12:53:40,372 | server.py:178 | evaluate_round 8: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:53:40,404 | server.py:192 | evaluate_round 8 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:53:40,405 | server.py:227 | fit_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:53:40,447 | server.py:241 | fit_round 9 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:40,458 | server.py:131 | fit progress: (9, 0.44020383819747594, {'accuracy': 0.7866666666666666}, 1.129337699999999)
DEBUG flwr 2024-07-11 12:53:40,459 | server.py:178 | evaluate_round 9: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:53:40,486 | server.py:192 | evaluate_round 9 received 5 results and 0 failures
DEBUG flwr 2024-07-11 12:53:40,487 | server.py:227 | fit_round 10: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:53:40,530 | server.py:241 | fit_round 10 received 5 results and 0 failures
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
D:\Anaconda3\envs\fl_py37\lib\site-packages\sklearn\base.py:444: UserWarning: X has feature names, but LogisticRegression was fitted without feature names
  f"X has feature names, but {self.__class__.__name__} was fitted without"
INFO flwr 2024-07-11 12:53:40,543 | server.py:131 | fit progress: (10, 0.44020475329461295, {'accuracy': 0.7866666666666666}, 1.2134400000000003)
DEBUG flwr 2024-07-11 12:53:40,543 | server.py:178 | evaluate_round 10: strategy sampled 5 clients (out of 5)
DEBUG flwr 2024-07-11 12:53:40,573 | server.py:192 | evaluate_round 10 received 5 results and 0 failures
INFO flwr 2024-07-11 12:53:40,573 | server.py:153 | FL finished in 1.2440216999999993
INFO flwr 2024-07-11 12:53:40,574 | app.py:225 | app_fit: losses_distributed [(1, 0.41107042878866196), (2, 0.40738531748453777), (3, 0.4053275426228841), (4, 0.4053286681572596), (5, 0.4053286780913671), (6, 0.4053286780913671), (7, 0.4053286780913671), (8, 0.4072604768332981), (9, 0.4068031885794231), (10, 0.4068031729686828)]
INFO flwr 2024-07-11 12:53:40,574 | app.py:226 | app_fit: metrics_distributed_fit {}
INFO flwr 2024-07-11 12:53:40,574 | app.py:227 | app_fit: metrics_distributed {}
INFO flwr 2024-07-11 12:53:40,574 | app.py:228 | app_fit: losses_centralized [(0, 0.6931471805599453), (1, 0.4309062369547472), (2, 0.43567787449586104), (3, 0.44566503187687334), (4, 0.445645019378396), (5, 0.44564502862649996), (6, 0.44564502863521266), (7, 0.4456450286352385), (8, 0.4456450286352386), (9, 0.44020383819747594), (10, 0.44020475329461295)]
INFO flwr 2024-07-11 12:53:40,574 | app.py:229 | app_fit: metrics_centralized {'accuracy': [(0, 0.4033333333333333), (1, 0.79), (2, 0.79), (3, 0.7866666666666666), (4, 0.7866666666666666), (5, 0.7866666666666666), (6, 0.7866666666666666), (7, 0.7866666666666666), (8, 0.7866666666666666), (9, 0.7866666666666666), (10, 0.7866666666666666)]}
>>>>>>>>>>>>> setting initial parameters
[[0. 0. 0. 0. 0. 0. 0. 0.]
 [0. 0. 0. 0. 0. 0. 0. 0.]]
[array([[0., 0., 0., 0., 0., 0., 0., 0.],
       [0., 0., 0., 0., 0., 0., 0., 0.]]), array([0., 0.])]
[array([[-1.89326641e+00,  9.31736036e-01,  7.74822653e-01,
         4.09116632e-01,  2.89378011e-01,  6.45529150e-04,
         1.74060176e-01,  5.33612254e-02]]), array([-0.16745865])]
[array([[-1.96107418,  0.92550622,  0.77622687,  0.35751438,  0.23434862,
        -0.03806502,  0.07384655,  0.02898432]]), array([0.11034723])]
[array([[-2.02029664,  0.99174705,  0.85246281,  0.43035612,  0.24978529,
         0.0133599 ,  0.05733874, -0.07043165]]), array([-0.06343647])]
[array([[-2.01956939,  0.99140269,  0.85233914,  0.43026975,  0.24944112,
         0.01350915,  0.05728606, -0.0704921 ]]), array([-0.06327064])]
[array([[-2.01957026,  0.99140243,  0.85233985,  0.43026977,  0.24943981,
         0.01350953,  0.05728627, -0.0704922 ]]), array([-0.06326923])]
[array([[-2.01957026,  0.99140243,  0.85233985,  0.43026977,  0.24943981,
         0.01350953,  0.05728627, -0.0704922 ]]), array([-0.06326923])]
[array([[-2.01957026,  0.99140243,  0.85233985,  0.43026977,  0.24943981,
         0.01350953,  0.05728627, -0.0704922 ]]), array([-0.06326923])]
[array([[-2.01957026,  0.99140243,  0.85233985,  0.43026977,  0.24943981,
         0.01350953,  0.05728627, -0.0704922 ]]), array([-0.06326923])]
[array([[-1.98236508,  0.98887372,  0.85432285,  0.39136341,  0.24586052,
         0.04435482,  0.07360801, -0.03154076]]), array([-0.10700631])]
[array([[-1.98235808,  0.98887073,  0.85430773,  0.391357  ,  0.24586605,
         0.044372  ,  0.07359727, -0.0315542 ]]), array([-0.10699669])]
>>>>>>>>>>>>>run time:
0.515625s
